dags/operators/task_sensor.py

```diff
@@ -0,0 +1,108 @@
+            dttm = context['execution_date'] - self.execution_delta
+        elif self.execution_date_fn:
+            dttm = self._handle_execution_date_fn(context=context)
+        else:
+            dttm = context['execution_date']
+
+        dttm_filter = dttm if isinstance(dttm, list) else [dttm]
+        serialized_dttm_filter = ','.join(
+            [datetime.isoformat() for datetime in dttm_filter])
+
+        self.log.info(
+            'Poking for %s.%s on %s ... ',
+            self.external_dag_id, self.external_task_id, serialized_dttm_filter
+        )
+
+        DM = DagModel
+        TI = TaskInstance
+        DR = DagRun
+        if self.check_existence:
+            dag_to_wait = session.query(DM).filter(
+                DM.dag_id == self.external_dag_id
+            ).first()
+
+            if not dag_to_wait:
+                raise AirflowException('The external DAG '
+                                       '{} does not exist.'.format(self.external_dag_id))
+            else:
+                if not os.path.exists(dag_to_wait.fileloc):
+                    raise AirflowException('The external DAG '
+                                           '{} was deleted.'.format(self.external_dag_id))
+
+            if self.external_task_id:
+                refreshed_dag_info = DagBag(dag_to_wait.fileloc).get_dag(self.external_dag_id)
+                if not refreshed_dag_info.has_task(self.external_task_id):
+                    raise AirflowException('The external task'
+                                           '{} in DAG {} does not exist.'.format(self.external_task_id,
+                                                                                 self.external_dag_id))
+        
+        # custom implementation to check for failed tasks
+        if self.external_task_id:
+            # .count() is inefficient
+            count_allowed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.allowed_states),
+                TI.execution_date.in_(dttm_filter),
+            ).scalar()
+
+            count_failed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.failed_states),
+                TI.execution_date.in_(dttm_filter),
+            ).scalar()
+        else:
+            # .count() is inefficient
+            count_allowed = session.query(func.count()).filter(
+                DR.dag_id == self.external_dag_id,
+                DR.state.in_(self.allowed_states),
+                DR.execution_date.in_(dttm_filter),
+            ).scalar()
+
+            count_failed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.failed_states),
```

Comment:
if you change self.failed_states to include upstream_failed and/or skipped, this line will need to change to compare DR.state with just the dag_state.FAILED